src/api/api.ts

```diff
@@ -0,0 +1,69 @@
+import { fetch } from "cross-fetch";
+import { type Result, Ok, Err } from "../types";
+import { hashSigningKey } from "../helpers/strings";
+import {
+  ErrorSchema,
+  type ErrorResponse,
+  StepsSchema,
+  type StepsResponse,
+  BatchSchema,
+  type BatchResponse,
+} from "./schema";
+
+interface InngestAPIConstructorOpts {
+  baseUrl?: string;
+  signingKey: string;
+}
+
+export class InngestAPI {
+  public readonly baseUrl: string;
+  private signingKey: string;
+
+  constructor({
+    baseUrl = "https://api.inngest.com",
+    signingKey,
+  }: InngestAPIConstructorOpts) {
+    this.baseUrl = baseUrl;
+    this.signingKey = signingKey;
+  }
+
+  private get hashedKey(): string {
+    return hashSigningKey(this.signingKey);
+  }
+
+  async getRunSteps(
+    runId: string
+  ): Promise<Result<StepsResponse, ErrorResponse>> {
+    const url = `${this.baseUrl}/v0/runs/${runId}/actions`;
+
+    return fetch(url, {
+      headers: { Authorization: `Bearer ${this.hashedKey}` },
+    }).then(async (resp) => {
+      const data: unknown = await resp.json();
```

Comment:
If `fetch()`, `resp.json()`, or `ErrorSchema.parse(data)` throws here, the function will throw instead of returning an `ErrorResponse`. Is this okay?

If we want to always return an `ErrorResponse`, it might be reasonable here (and in `getRunBatch()`) to wrap the entire call in a `try`/`catch`.

Alternatively you could just add a `.catch()` to the end of the `fetch()` call, but you won't have access to the `data` to parse it.

Reply:
I'll do a catch at the end. if it fails, data won't be available anyways.